File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int max = 0;
        int res = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i : nums){
            map.put(i, map.getOrDefault(i, 0) + 1);
            if(map.get(i) > max){
                res = i;
                max = map.get(i);
            }
        }
        return res;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hashMap = {}
        maxx = 0
        res = 0
        for index in nums:
            hashMap[index] = hashMap.get(index, 0) + 1
            if hashMap.get(index) > maxx:
                maxx = hashMap.get(index)
                res = index
        print(hashMap)
        return res

